nautobot/extras/models/mixins.py

```diff
@@ -28,6 +28,23 @@ def dynamic_groups(self):
 
         return self._dynamic_group_queryset
 
+    def get_dynamic_groups_url(self):
+        """Return the dynamic groups URL for a given instance."""
+        route = get_route_for_model(self, "dynamicgroups")
+
+        # Iterate the pk-like fields and try to get a URL, or return None.
+        fields = ["pk", "slug"]
+        for field in fields:
+            if not hasattr(self, field):
+                continue
+
+            try:
+                return reverse(route, kwargs={field: getattr(self, field)})
+            except NoReverseMatch:
+                continue
```

Comment:
There's that pattern again. We should probably factor this one into a helper at some point.